File: easy school/registration/frmparents.vb
Public Class mdyfrmparents
    Dim DATA As New database
    Public id, names, tel, email, work, employer, residents As String
    Private Sub TextBox7_TextChanged(sender As Object, e As EventArgs)

    End Sub

    Function checkblank(ByVal sender As Object) As Boolean
        If sender.text.ToString.Length = 0 Then
            MsgBox("can't be blank", MsgBoxStyle.Information, "")
            ' sender.focus = True
            Return False
        End If
    End Function
    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Try
            If checkblank(TextBox13) = False Then
                Exit Sub
            ElseIf checkblank(TextBox7) = False Then
                Exit Sub
            ElseIf checkblank(TextBox7) = False Then
                Exit Sub
            ElseIf checkblank(TextBox11) = False Then
                Exit Sub
            ElseIf checkblank(TextBox21) = False Then
                Exit Sub
            ElseIf checkblank(TextBox8) = False Then
                Exit Sub
            ElseIf checkblank(TextBox13) = False Then
                Exit Sub
            ElseIf checkblank(TextBox12) = False Then
                Exit Sub
            Else
                id = TextBox13.Text
                names = TextBox7.Text
                tel = TextBox11.Text
                email = TextBox21.Text
                work = TextBox8.Text
                employer = TextBox9.Text
                residents = TextBox12.Text
                DATA.add("INSERT INTO " & Me.Text & "  VALUES (" & id & ", '" & names & "', '" & tel & "', '" & email & "', '" & work & "', '" & employer & "', '1');")
                Me.Close()
            End If
        Catch ex As Exception

        End Try
    End Sub

    Private Sub frmparents_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Label1.Text = Me.Text.ToUpper & "'S DETAILS"
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Me.Close()
    End Sub

    Private Sub TextBox7_TextChanged_1(sender As Object, e As EventArgs) Handles TextBox7.TextChanged
        Dim val As New validate
        val.alphas(sender.text)
    End Sub

    Private Sub TextBox13_KeyPress(sender As Object, e As KeyPressEventArgs) Handles TextBox13.KeyPress
        Dim val As New validate
        val.digits(sender, e)
    End Sub

    Private Sub TextBox13_TextChanged(sender As Object, e As EventArgs) Handles TextBox13.TextChanged


    End Sub

    Private Sub TextBox11_KeyPress(sender As Object, e As KeyPressEventArgs) Handles TextBox11.KeyPress
        Dim val As New validate
        val.digits(sender, e)
    End Sub

    Private Sub TextBox11_TextChanged(sender As Object, e As EventArgs) Handles TextBox11.TextChanged

    End Sub

    Private Sub TextBox21_TextChanged(sender As Object, e As EventArgs) Handles TextBox21.TextChanged

    End Sub

    Private Sub TextBox21_Validating(sender As Object, e As System.ComponentModel.CancelEventArgs) Handles TextBox21.Validating
        Dim val As New validate
        val.emails(sender)
    End Sub

    Private Sub TextBox8_TextChanged(sender As Object, e As EventArgs) Handles TextBox8.TextChanged
        Dim val As New validate
        val.alphas(sender.text)
    End Sub

    Private Sub TextBox9_TextChanged(sender As Object, e As EventArgs) Handles TextBox9.TextChanged
        Dim val As New validate
        val.alphas(sender.text)
    End Sub

    Private Sub TextBox12_TextChanged(sender As Object, e As EventArgs) Handles TextBox12.TextChanged
        Dim val As New validate
        val.alphas(sender.text)
    End Sub
End Class

File: easy school.ConvertedToC#/registration/frmparents.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
namespace easy_school
{
	public partial class mdyfrmparents
	{
		database DATA = new database();
		public string id;
		public string names;
		public string tel;
		public string email;
		public string work;
		public string employer;
		public string residents;

		private void TextBox7_TextChanged(object sender, EventArgs e)
		{
		}

		public bool checkblank(object sender)
		{
			if (sender.text.ToString.Length == 0) {
				Interaction.MsgBox("can't be blank", MsgBoxStyle.Information, "");
				// sender.focus = True
				return false;
			}
		}
		private void Button2_Click(object sender, EventArgs e)
		{
			try {
				if (checkblank(TextBox13) == false) {
					return;
				} else if (checkblank(TextBox7) == false) {
					return;
				} else if (checkblank(TextBox7) == false) {
					return;
				} else if (checkblank(TextBox11) == false) {
					return;
				} else if (checkblank(TextBox21) == false) {
					return;
				} else if (checkblank(TextBox8) == false) {
					return;
				} else if (checkblank(TextBox13) == false) {
					return;
				} else if (checkblank(TextBox12) == false) {
					return;
				} else {
					id = TextBox13.Text;
					names = TextBox7.Text;
					tel = TextBox11.Text;
					email = TextBox21.Text;
					work = TextBox8.Text;
					employer = TextBox9.Text;
					residents = TextBox12.Text;
					DATA.@add(ref "INSERT INTO " + this.Text + "  VALUES (" + id + ", '" + names + "', '" + tel + "', '" + email + "', '" + work + "', '" + employer + "', '1');");
					this.Close();
				}

			} catch (Exception ex) {
			}
		}

		private void frmparents_Load(object sender, EventArgs e)
		{
			Label1.Text = this.Text.ToUpper() + "'S DETAILS";
		}

		private void Button1_Click(object sender, EventArgs e)
		{
			this.Close();
		}

		private void TextBox7_TextChanged_1(object sender, EventArgs e)
		{
			validate val = new validate();
			val.alphas(sender.text);
		}

		private void TextBox13_KeyPress(object sender, KeyPressEventArgs e)
		{
			validate val = new validate();
			val.digits(sender, e);
		}


		private void TextBox13_TextChanged(object sender, EventArgs e)
		{

		}

		private void TextBox11_KeyPress(object sender, KeyPressEventArgs e)
		{
			validate val = new validate();
			val.digits(sender, e);
		}


		private void TextBox11_TextChanged(object sender, EventArgs e)
		{
		}


		private void TextBox21_TextChanged(object sender, EventArgs e)
		{
		}

		private void TextBox21_Validating(object sender, System.ComponentModel.CancelEventArgs e)
		{
			validate val = new validate();
			val.emails(sender);
		}

		private void TextBox8_TextChanged(object sender, EventArgs e)
		{
			validate val = new validate();
			val.alphas(sender.text);
		}

		private void TextBox9_TextChanged(object sender, EventArgs e)
		{
			validate val = new validate();
			val.alphas(sender.text);
		}

		private void TextBox12_TextChanged(object sender, EventArgs e)
		{
			validate val = new validate();
			val.alphas(sender.text);
		}
		public mdyfrmparents()
		{
			Load += frmparents_Load;
			InitializeComponent();
		}
	}
}
